### DISCUSSION TITLE
What is type conversion in parameter for?

### DISCUSSION THREAD

**Discussion by harrypunk** [bevyengine/bevy#1918] [Q&A] [answered]
Hi,
I was reading the source code and came across the type conversion in function signatures:
https://github.com/bevyengine/bevy/blob/97d8e4e1793ede3df8c77ed44736e800b38ff7a4/crates/bevy_ecs/src/system/into_system.rs
```rust
pub struct FunctionSystem<In, Out, Param, Marker, F>
where
    Param: SystemParam,
{
    config: Option<<Param::Fetch as SystemParamState>::Config>,
}
```
I just wonder what the conversion does here, Why not ```SystemParamState::Config```?

**Comment by bjorn3** [comment, marked as answer]
The type `Param::Fetch` implements the trait `SystemParamState` which has an associated type `Config`. The `as SystemParamState` indicates that the associated type `Config` should be looked for in the implementation of `SystemParamState` for `Param::Fetch`.
